src/service/resource.js

```diff
@@ -614,7 +614,7 @@ export class Resource {
       dev().fine(TAG, 'layout complete:', this.debugid);
     } else {
       dev().fine(TAG, 'loading failed:', this.debugid, opt_reason);
-      return Promise.reject(opt_reason);
+      return Promise.reject(dev().createError('loading failed', opt_reason));
```

Comment:
Many layout problems could be a result of "user" error, such as misconfiguring an AMP element or problems executing XHRs (404, etc). I think this is a prevalent case actually.